lib/board.rb

```diff
@@ -41,4 +41,83 @@ def update_grid(location, mark)
     end
   end
 
+  def full_board?(board_grid)
+    board_grid.uniq.length == 2
+  end
+
+  def get_all_move_combos
+    combinations = [get_all_rows, get_all_columns, get_diagonals]
+
+    combinations.flatten(1)
+  end
+
+  def get_all_rows
+    horizontal_moves = []
+    start = 0
+    row_length = @size
+    target_row = @grid[start, row_length]
+
+    while start < @grid.length
+      target_row = @grid[start, row_length]
+      horizontal_moves.push(target_row)
+      start = start + row_length
+    end
+
+    horizontal_moves
+  end
+
+  def get_all_columns
+    vertical_moves = []
+    pointer = 0
+    col_length = @size
+
+    while pointer < col_length
+      col = []
+      @grid.each_with_index do|value, index|
+        if (index - pointer) % col_length == 0
+          col.push(value)
+        end
+      end
+
+      pointer = pointer + 1
+      vertical_moves.push(col)
+      col = []
+    end
+
+    vertical_moves
+  end
+
+  def get_diagonals
+    [get_main_diagonal, get_antidiagonal]
+  end
+
+  def get_antidiagonal
+    diagonal_moves = []
+    diagonal_length = @size
+
+    @grid.each_with_index do|value, index|
+      if (diagonal_length - 1 - index) % (diagonal_length - 1) == 0 && index != 0 && index != @grid.length - 1
```

Comment:
I would abstract this into a private method for readability.

Reply:
done!